### ISSUE TITLE
[BUG] SemanticChunker and SDPMChunker Truncates Last Sentence

### ISSUE THREAD

**Issue by philipchung** [chonkie-ai/chonkie#106]
**Describe the bug**
Semantic Chunker truncates the last sentence and does not return it in the final chunk. SDPMChunker relies on the same logic and has the same issue. This appears to be an indexing issue where the last sentence is left out of the last group. This issue occurs in `chonkie 0.3.0`

**To Reproduce**
```python
from chonkie.chunker import SemanticChunker

chunker = SemanticChunker(
    embedding_model="minishlab/potion-base-8M",
    mode="window",
    threshold="auto",
    chunk_size=10,
    similarity_window=1,
    min_sentences=1,
)
text = "Alice ate an apple. Bob ate a banana. Charlie ate a cherry. David ate a date."

chunks = chunker.chunk(text)

# The below methods are used in Semantic Chunker's internal logic
sentences = chunker._prepare_sentences(text)
similarity_threshold = chunker._calculate_similarity_threshold(sentences)
similarities = chunker._compute_pairwise_similarities(sentences)
split_indices = chunker._get_split_indices(similarities, chunker.similarity_threshold)
groups = [sentences[split_indices[i] : split_indices[i + 1]] for i in range(len(split_indices) - 1)]
```

```python
# Inspect Chunks
for chunk in chunks:
    print(f"Chunk Text: {chunk.text}")

# Output:
# Chunk Text: Alice ate an apple.
# Chunk Text:  Bob ate a banana.
# Chunk Text:  Charlie ate a cherry.
```

```python
# Inspect Chunked Sentences
for sentence in sentences:
    print(
        f"Sentence Text: {sentence.text}\n"
        f"Start Index: {sentence.start_index}\n"
        f"End Index: {sentence.end_index}\n"
    )

# Output:
# Sentence Text: Alice ate an apple.
# Start Index: 0
# End Index: 19
#
# Sentence Text:  Bob ate a banana.
# Start Index: 19
# End Index: 37

# Sentence Text:  Charlie ate a cherry.
# Start Index: 37
# End Index: 59

# Sentence Text:  David ate a date.
# Start Index: 59
# End Index: 77
```

```python
print(
    f"Num Chunks: {len(chunks)}\n"
    f"Num Sentences: {len(sentences)}\n"
    f"Num Similarities: {len(similarities)}\n"
    f"Num Split Indices: {len(split_indices)}\n"
    f"Num Groups: {len(groups)}\n"
)

# Output:
# Num Chunks: 3
# Num Sentences: 4
# Num Similarities: 3
# Num Split Indices: 3
# Num Groups: 2
```

```python
# Inspect Group 0
[sentence.text for sentence in groups[0]]
# Output:
# ['Alice ate an apple.', ' Bob ate a banana.']

# Inspect Group 1
[sentence.text for sentence in groups[1]]
# Output:
# [' Charlie ate a cherry.']. <-- this group is missing the last sentence
```

**Expected behavior**
Last sentence should not be dropped from the final group.

**Comment by bhavnicksm** [comment]
Hey @philipchung! 👋

Thanks for opening an issue and for the detailed error reproduction code~ 

I'm taking this issue up right now, will recreate it, and understand what's going on. Will ping when I make progress on this...

Thanks! 😊

**Comment by bhavnicksm** [comment]
Hey @philipchung!

Just merged a patch for the issue (#112) and verified your test case as well. Thank you for your patience. 

I will add it to the source soon, so you can verify it. And will be released with chonkie's next feature release as well :)

Thanks 😊
